pyaedt/modeler/cad/Primitives.py

```diff
@@ -275,45 +587,4908 @@ def version(self):
+        return True, list(added_objs)
+
+    def _duplicate_added_objects_tuple(self):
+        added_objects = self.add_new_objects()
+        if added_objects:
+            return True, added_objects
+        else:
+            return False, []
+
+    def _duplicate_added_components_tuple(self):
+        added_component = self.add_new_user_defined_component()
+        if added_component:
+            return True, added_component
+        else:
+            return False, []
+
+    @pyaedt_function_handler()
+    def duplicate_along_line(
+        self,
+        objid,
+        vector,
+        nclones=2,
+        attachObject=False,
+        is_3d_comp=False,
+        duplicate_assignment=True,
+    ):
+        """Duplicate a selection along a line.
+
+        Parameters
+        ----------
+        objid : list, str, int, :class:`pyaedt.modeler.Object3d.Object3d`
+            Name or ID of the object.
+        vector : list
+            List of ``[x1,y1,z1]`` coordinates or the Application.Position object for
+            the vector.
+        attachObject : bool, optional
+            The default is ``False``.
+        nclones : int, optional
+            Number of clones. The default is ``2``.
+        is_3d_comp : bool, optional
+            If True, the method will try to return the duplicated list of 3dcomponents. The default is ``False``.
+        duplicate_assignment : bool, optional
+            If True, the method duplicates selection assignments. The default value is ``True``.
+
+        Returns
+        -------
+        tuple
+
+        References
+        ----------
+
+        >>> oEditor.DuplicateAlongLine
+        """
+        selections = self.convert_to_selections(objid)
+        Xpos, Ypos, Zpos = self._pos_with_arg(vector)
+
+        vArg1 = ["NAME:Selections", "Selections:=", selections, "NewPartsModelFlag:=", "Model"]
+        vArg2 = ["NAME:DuplicateToAlongLineParameters"]
+        vArg2.append("CreateNewObjects:="), vArg2.append(not attachObject)
+        vArg2.append("XComponent:="), vArg2.append(Xpos)
+        vArg2.append("YComponent:="), vArg2.append(Ypos)
+        vArg2.append("ZComponent:="), vArg2.append(Zpos)
+        vArg2.append("Numclones:="), vArg2.append(str(nclones))
+        vArg3 = ["NAME:Options", "DuplicateAssignments:=", duplicate_assignment]
+        self.oeditor.DuplicateAlongLine(vArg1, vArg2, vArg3)
+        if is_3d_comp:
+            return self._duplicate_added_components_tuple()
+        if attachObject:
+            return True, []
+        return self._duplicate_added_objects_tuple()
+
+    @pyaedt_function_handler()
+    def thicken_sheet(self, objid, thickness, bBothSides=False):
+        """Thicken the sheet of the selection.
+
+        Parameters
+        ----------
+        objid : list, str, int, :class:`pyaedt.modeler.Object3d.Object3d`
+            Name or ID of the object.
+        thickness : float, str
+            Amount to thicken the sheet by.
```

Comment:
Does a unit need to be mentioned here. (For example, "Amount in mm to thicken the sheet by."